resin/main.go

```diff
@@ -115,16 +184,33 @@ func main() {
 		os.Exit(2)
 	}
 
-	// if shell is relative, prepend with dir
-	if viper.Get("shell").(string)[0] != '/' {
-		viper.Set("shell", path.Join(viper.GetString("dir"), viper.GetString("shell")))
+	// if paths are relative, prepend with dir and verify files exist
+	fix_path_check_exists := func(key string) {
+		if viper.Get(key).(string)[0] != '/' {
```

Comment:
This line will panic if the value is not a `string`
https://golang.org/ref/spec#Type_assertions